source/plugins/ruby/in_containerinventory.rb

```diff
@@ -63,10 +64,16 @@ def enumerate
       $log.info("in_container_inventory::enumerate : Begin processing @ #{Time.now.utc.iso8601}")
       if ExtensionUtils.isAADMSIAuthMode()
         $log.info("in_container_inventory::enumerate: AAD AUTH MSI MODE")
-        if @tag.nil? || !@tag.start_with?(Constants::EXTENSION_OUTPUT_STREAM_ID_TAG_PREFIX)
-          @tag = ExtensionUtils.getOutputStreamId(Constants::CONTAINER_INVENTORY_DATA_TYPE)
+        useFromCache = true
+        if !KubernetesApiClient.isDCRStreamIdTag(@tag) || (DateTime.now.to_time.to_i - @agentConfigRefreshTracker).abs >= Constants::AGENT_CONFIG_REFRESH_INTERVAL_SECONDS
```

Comment:
All of this changed code as well is same in all the files. Is there a way to have a function for this so next time we want to change anything/find bugs we dont have to change in all the plugins ?

Reply:
> All of this changed code as well is same in all the files. Is there a way to have a function for this so next time we want to change anything/find bugs we dont have to change in all the plugins ?

done.